pkg/plugins/external/api.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package external
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+
+	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
+	"sigs.k8s.io/kubebuilder/v3/pkg/model/resource"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/v3/pkg/plugin/external"
+)
+
+var _ plugin.CreateAPISubcommand = &createAPISubcommand{}
+
+const (
+	defaultAPIVersion          = "v1alpha1"
+	defaultCreateAPISubcommand = "create api"
+)
+
+type createAPISubcommand struct {
+	Path string
+	Args []string
+}
+
+func (p *createAPISubcommand) InjectResource(*resource.Resource) error {
+	// Do nothing since resource flags are passed to the external plugin directly.
+	return nil
+}
+
+func (p *createAPISubcommand) Scaffold(fs machinery.Filesystem) error {
+	req := external.PluginRequest{
+		APIVersion: defaultAPIVersion,
+	}
+
+	req.Universe = map[string]string{}
+
+	if req.Command == "" {
+		if len(p.Args) == 0 {
+			return fmt.Errorf("internal error: invalid length of arguments for %q subcommand", defaultCreateAPISubcommand)
```

Comment:
Seems weird to have `internal error:` in the message. 
```suggestion
			return fmt.Errorf("invalid length of arguments for %q subcommand", defaultCreateAPISubcommand)
```